### PR TITLE
New vite based application runtime

### PR DESCRIPTION
**Pull request by Janpot**
Use vite to build the Toolpad applications:

* Allows using installed node modules in Toolpad applications
* Faster production runtime that doesn't pull in every icon and every MUI component
* Allows running react in dev mode in the canvas to allow for richer error messages
* Added behind a `--viteRuntime` CLI flag to allow canary testing

Additional fixes:

* canvas overlay disappeared when the canvas iframe reloaded, added a fix for that
* store test results in CI for debuggability
* fix `/preview/pages` not redirecting to the first page
* Remove `next/image` usage
* Align some logs to the format throughout the app
* `dragNewComponentToAppCanvas` was exiting before the drag was completed, resulting in failures if the test wasn't waiting on anything afterwards
* Recreate yarn.lock

I can open a separate PR for those if you want, but most are needed for this PR as well

### CODE REVIEW

File: test/models/ToolpadEditor.ts [hunk 1]
```diff
@@ -144,23 +144,25 @@ export class ToolpadEditor {
     await this.page.mouse.up();
   }
 
+  componentCatalogItem(name: string): Locator {
```
Comment (apedroferreira) [inline test/models/ToolpadEditor.ts:147]: nit: maybe `getComponentCatalogItem`?

File: test/models/ToolpadEditor.ts [hunk 2]
```diff
@@ -144,23 +144,25 @@ export class ToolpadEditor {
...
     const moveTargetX = targetBoundingBox!.x + targetBoundingBox!.width / 2;
     const moveTargetY = targetBoundingBox!.y + targetBoundingBox!.height / 2;
 
-    const sourceLocator = this.page.getByTestId('component-catalog').getByRole('button', {
-      name: componentName,
-    });
+    const sourceLocator = this.componentCatalogItem(componentName);
 
-    this.dragToAppCanvas(sourceLocator, moveTargetX, moveTargetY);
+    await this.dragToAppCanvas(sourceLocator, moveTargetX, moveTargetY);
```
Comment (apedroferreira) [inline test/models/ToolpadEditor.ts:165]: 👍

**Review by apedroferreira — APPROVED** [review]

**Comment by apedroferreira** [comment]
Great feature! 
Hopefully Vite makes the development experience a bit faster too since it's more important in the new direction.

**Review by apedroferreira — COMMENTED** [review]
Wait, I had checked only the code but tried to run it and got:

Screenshot 2023-04-19 at 23 13 29

**Comment by Janpot** [comment]
> Wait, I had checked only the code but tried to run it and got:

Currently, to try out the new functionality you'll have to run the toolpad cli with `--viteRuntime` flag. Looks like I had a problem with the old mode. Fixed and running the tests in old mode to make sure it still works

File: test/models/ToolpadEditor.ts (hunk 1 above)
Comment (Janpot) [reply test/models/ToolpadEditor.ts:147]: sure, I tried to be consistent with `hierarchyItem` below

**Review by apedroferreira — APPROVED** [review]
`yarn dev` works now!
Also I've tried a custom component using `recharts` and it worked great.

Screenshot 2023-04-20 at 09 49 17

**Comment by Janpot** [comment]
Yes, I've added an integration test for `recharts`.

**Pull request merged**